tests/plugins/test_koji_import.py

```diff
@@ -1295,43 +1295,39 @@ def test_koji_import_add_help(self, tmpdir, os_env, expect_result):
         elif expect_result in ['skip', 'unknown_status']:
             assert 'help' not in image.keys()
 
-    @pytest.mark.parametrize('has_pulp_sync', (True, False))
-    @pytest.mark.parametrize('version', [
-        # no pulp plugin used
-        None,
-        # V1-only image
-        ["application/json"],
-        # V1+V2schema1 image, i.e. what we are building today
-        ["application/json", "application/vnd.docker.distribution.manifest.v1+json"],
-        # V1+V2schema2 image
-        ["application/json", "application/vnd.docker.distribution.manifest.v1+json",
-         "application/vnd.docker.distribution.manifest.v2+json"],
-        # manifest lists, as well as compatibility formats back to V1
-        ["application/json", "application/vnd.docker.distribution.manifest.v1+json",
-         "application/vnd.docker.distribution.manifest.v2+json",
-         "application/vnd.docker.distribution.manifest.list.v2+json"],
-        # manifest lists, with compatibility formats back to V2schema1, without V1
-        ["application/vnd.docker.distribution.manifest.v1+json",
-         "application/vnd.docker.distribution.manifest.v2+json",
-         "application/vnd.docker.distribution.manifest.list.v2+json"],
+    @pytest.mark.parametrize(('worker', 'pulp_sync', 'expected'), [
+        (None, None, None),
+        (["application/json"], None, ["application/json"]),
+        (["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json"], None,
+         ["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json"]),
+        (["application/json",
+          "application/vnd.docker.distribution.manifest.v1+json",
+          "application/vnd.docker.distribution.manifest.v2+json"],
+         ["application/vnd.docker.distribution.manifest.v1+json",
+          "application/vnd.docker.distribution.manifest.v2+json",
+          "application/vnd.docker.distribution.manifest.list.v2+json"],
+         ["application/json",
+          "application/vnd.docker.distribution.manifest.list.v2+json",
+          "application/vnd.docker.distribution.manifest.v1+json",
+          "application/vnd.docker.distribution.manifest.v2+json"]),
```

Comment:
This needs to be easier to read so we can clearly see what we're testing/expecting.

What about this sort of layout?:
```
@pytest.mark.parametrize(('config', 'expected'), [
    # API V1-only build, arrangement version < 4
    ({'pulp_push': True,
      'pulp_pull_in_worker': True,
      'pulp_pull_in_orchestrator': False,
      'pulp_pull_media_types': []},
      ["application/json"]),

    ...
    # API V1+V2 build, arrangement version >= 4, schema2-capable Pulp
    ({'pulp_push': True,
       'pulp_pull_in_worker': False,
       'pulp_pull_in_orchestrator': True,
       'pulp_pull_media_types': [
           "application/vnd.docker.distribution.manifest.v1+json",
           "application/vnd.docker.distribution.manifest.v2+json",
           "application/vnd.docker.distribution.manifest.v2.list+json",
       ]},
      ["application/json",
       "application/vnd.docker.distribution.manifest.v1+json",
       "application/vnd.docker.distribution.manifest.v2+json",
       "application/vnd.docker.distribution.manifest.v2.list+json"]),
...
```

Reply:
Agree, its a bit excessive. Is it worth setting `pulp_pull: {'schema1': True, 'schema2': True, 'list': False}`?

Reply:
Yes, that sounds even better.

Reply:
Implemented